### PR TITLE
TD-1470 Fixed issue in generating verify email link by changing the p…

### PR DESCRIPTION
**Pull request by manishagarwalnhs**
…arameters passed for getting hash id and generate mail content

### JIRA link
https://hee-tis.atlassian.net/browse/TD-1470

### Description
Points addressed in this Commit :
1) Fixed issue in generating verify email link by changing the parameters passed for getting hash id and generate mail content in controller action.

### Screenshots
This is backend change so no screenshot available.

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [ ] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
@manishagarwalnhs, this change doesn't seem to make any difference to the email generated. We still pick up the existing `DelegateAccounts.RegistrationConfirmationHash` and the `DelegateAccounts.RegistrationConfirmationHashCreationDateTime` does not change.

Given that the link is time-limited, if the administrator generates the email after the time limit has expired, the learner won't be able to use it. We want to generate a new hash and reset the creation time in the same way that happens when the "Send welcome email" button is clicked. 

Because we are simply including a link to open an email, we can't achieve this, though, so we should load the RegistrationConfirmationHashCreationDateTime into the view model and only show the link if it is more than 1 day in the future. Ideally we should also calculate the number of days and include that detail in the email, though that might be tricky.

**Comment by manishagarwalnhs** [comment]
"Open welcome email" and "Open verification email" links will only be shown if there is more than 1 day left in its expiry.

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/ViewDelegateController.cs [hunk 1]
```diff
@@ -79,18 +81,25 @@ public IActionResult Index(int delegateId)
                 model.DelegateInfo.Email = null;
 
             var baseUrl = config.GetAppRootPath();
-            if ((model.DelegateInfo?.IsActive ?? false) && (model.DelegateInfo.RegistrationConfirmationHash != null))
+            IClockUtility clockUtility = new ClockUtility();
+            if ((model.DelegateInfo?.IsActive ?? false) && (model.DelegateInfo.RegistrationConfirmationHash != null
+                && model.DelegateInfo.RegistrationConfirmationHashCreationDateTime?.AddDays(2) > clockUtility.UtcNow)
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/ViewDelegateController.cs:86]: The validity period will be moved to appsettings in https://hee-tis.atlassian.net/browse/TD-1863 please connect with @rshrirohit to make sure this is picked up here.

**Review by kevwhitt-hee — COMMENTED** [review]
Happy to approve this once the comment re https://hee-tis.atlassian.net/browse/TD-1863 has been acknowledged.

We should add a ticket to specify the number of days the link will remain valid in the email text too, @rshrirohit, since we could be sending an email stating it is valid for 14 days when in fact it is only valid for 2 or 3.

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/ViewDelegateController.cs (hunk 1 above)
Comment (manishagarwalnhs) [reply DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/ViewDelegateController.cs]: @kevwhitt-hee This is now not required anymore I believe because we have reverted back the changes.

File: DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserDataService.cs [hunk 2]
```diff
@@ -125,6 +125,7 @@ public partial class UserDataService
                 da.UserID,
                 da.CentreSpecificDetailsLastChecked,
                 da.RegistrationConfirmationHash,
+                da.RegistrationConfirmationHashCreationDateTime,
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserDataService.cs:128]: We need to revert the changes that bring the `RegistrationConfirmationHashCreationDateTime` into the view model since we no longer do anything with it.

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
See code comments, need to revert changes that bring the `RegistrationConfirmationHashCreationDateTime` into the model since it is not used any more.

File: DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserDataService.cs (hunk 2 above)
Comment (manishagarwalnhs) [reply DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserDataService.cs]: This is addressed.
